### PR TITLE
Feat/1084-prompt-for-wallet-upgrade

### LINKED ISSUE #1084: Prompt people to upgrade to new versions of the wallet when there is a new version out
FE can query the releases of vega wallet through the github API. If there is a new version we can show a message prompting users to upgrade to the new shinny version. As we are pre 1.0.0 and there can be breaking changes as per semver this would be a good shout.

https://api.github.com/repos/vegaprotocol/vegawallet/releases

and find the latest non-prerelease

### PR DESCRIPTION
**Pull request by pgriscti**
closes #1084

### CODE REVIEW

File: src/components/vega-wallet/new-wallet-available.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const [newVersionAvailable, setNewVersionAvailable] =
+    React.useState<string | null>(null);
+  const [latestVersionLocation, setLatestVersionLocation] =
+    React.useState<string>();
+
+  React.useEffect(() => {
+    const getWalletLatest = async () => {
+      const [latestVersion, location] =
+        await vegaWalletService.getLastestVersion();
+      const [, currentVersion] = await vegaWalletService.getVersion();
```
Comment (dexturr) [inline src/components/vega-wallet/new-wallet-available.tsx:19]: Let's do these in parallel please

File: src/lib/vega-wallet/vega-wallet-service.ts [hunk 2]
```diff
@@ -176,6 +177,29 @@ export class VegaWalletService implements IVegaWalletService {
...
+      // find the first non-prelease
+      for (let walletVerison of json) {
+        if (!walletVerison.prerelease) {
+          const location = Links.WALLET_RELEASES_LATEST + walletVerison.name;
+          return [walletVerison.name, location];
+        }
+      }
+      return [null, null];
+    } catch (err) {
+      return [null, null];
```
Comment (dexturr) [inline src/lib/vega-wallet/vega-wallet-service.ts:199]: add sentry report if e are getting errors from this.

File: src/lib/vega-wallet/vega-wallet-service.ts [hunk 3]
```diff
@@ -176,6 +177,29 @@ export class VegaWalletService implements IVegaWalletService {
     return semver.satisfies(version, MINIMUM_WALLET_VERSION);
   }
 
+  async getLastestVersion() {
```
Comment (mattrussell36) [inline src/lib/vega-wallet/vega-wallet-service.ts:180]: @dexturr @pgriscti I'm wondering if this should be in the wallet service at all? After all we aren't really interacting with the wallet here, but the github API 🤔

Comment (mattrussell36) [reply src/lib/vega-wallet/vega-wallet-service.ts:180]: I'm leaning towards it just living in the component it gets used in or perhaps in a file in lib

File: src/components/vega-wallet/new-wallet-available.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      try {
+        const res = await fetch(Links.WALLET_RELEASES_HISTORY, {});
+        const json = await res.json();
+
+        // find the first non-prelease
+        for (let walletVerison of json) {
+          if (!walletVerison.prerelease) {
+            const location = Links.WALLET_RELEASES_LATEST + walletVerison.name;
+            const latestVersion = walletVerison.name;
+            const [, currentVersion] = await vegaWalletService.getVersion();
```
Comment (mattrussell36) [inline src/components/vega-wallet/new-wallet-available.tsx:29]: Can we get the wallet version once outside of the loop?

File: src/components/vega-wallet/new-wallet-available.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+import { Links } from "../../config/links";
+import { vegaWalletService } from "../../lib/vega-wallet/vega-wallet-service";
+
+export const NewWalletAvailable = () => {
+  const { t } = useTranslation();
+  const [newVersionAvailable, setNewVersionAvailable] = React.useState<
+    string | null
+  >(null);
+  const [latestVersionLocation, setLatestVersionLocation] =
```
Comment (mattrussell36) [inline src/components/vega-wallet/new-wallet-available.tsx:15]: Minor nitpick: Can we make this consistent to the state above so it expects string | null and is initialized with null

**Review by mattrussell36 — CHANGES_REQUESTED** [review]
Looks apart from getting the current wallet version in the loop. Can we move that outside of the loop

File: src/components/vega-wallet/new-wallet-available.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    React.useState<string>();
+
+  React.useEffect(() => {
+    const getWalletLatest = async () => {
+      try {
+        const res = await fetch(Links.WALLET_RELEASES_HISTORY, {});
```
Comment (mattrussell36) [inline src/components/vega-wallet/new-wallet-available.tsx:21]: No need to pass the empty object here

File: src/components/vega-wallet/new-wallet-available.tsx (hunk 4 above)
Comment (pgriscti) [reply src/components/vega-wallet/new-wallet-available.tsx:29]: it's not needed outside the loop.

Comment (mattrussell36) [reply src/components/vega-wallet/new-wallet-available.tsx:30]: For each iteration of the loop `vegaWalletService.getVersion()` will return the same thing every time so I think its unnecessary to make a network request to the wallet every time.

Something like the below would be better 🕊️ 
```
const [, currentVersion] = await vegaWalletService.getVersion();

for (let walletVersion of json) {
  ...
  if (latestVersion !== currentVersion) {
    ... // setstate
  }
}
```

Comment (pgriscti) [reply src/components/vega-wallet/new-wallet-available.tsx:30]: Oh, I see, I thought you were pointing out that `latestVersion` was not available once the loop ended.

**Review by dexturr — APPROVED** [review]

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**